### PR TITLE
Small code cleanups

### PR DESCRIPTION
**Pull request by pervazea**
- remove unnecessary UNUSED_ATTRIBUTE designations
- LOG_ERROR -> LOG_WARNING for SSL startup

### CODE REVIEW

File: src/storage/data_table.cpp [hunk 1]
```diff
@@ -821,6 +821,8 @@ bool DataTable::CheckForeignKeyConstraints(
   return true;
 }
 
+
```
Comment (pmenon) [inline src/storage/data_table.cpp]: Remove added lines.

File: src/network/peloton_server.cpp [hunk 2]
```diff
@@ -125,7 +125,7 @@ void PelotonServer::SSLInit() {
   //TODO(Yuchen): change this.
   //load trusted CA certificates (peer authentication)
   if (SSL_CTX_load_verify_locations(ssl_context, certificate_file_.c_str(), nullptr) != 1) {
-    LOG_ERROR("Exception when loading root_crt!");
+    LOG_WARN("Exception when loading root_crt!");
```
Comment (pmenon) [inline src/network/peloton_server.cpp]: @ChTimTsubasa Are these actually errors? Can we safely downgrade them to warnings?

Comment (ChTimTsubasa) [reply src/network/peloton_server.cpp]: I consulted @haoxianghua and @dengyuchenkit today and this line can be downgraded to warning with further information. I would address the richer information as well as a fix for the log throwing here in an upcoming PR.

File: src/network/peloton_server.cpp [hunk 3]
```diff
@@ -120,7 +120,7 @@ void PelotonServer::SSLInit() {
   // load trusted CA certificates (peer authentication)
   if (SSL_CTX_load_verify_locations(ssl_context, certificate_file_.c_str(),
                                     nullptr) != 1) {
-    LOG_ERROR("Exception when loading root_crt!");
+    LOG_WARN("Exception when loading root_crt!");
```
Comment (pmenon) [inline src/network/peloton_server.cpp]: Do we want to add a more descriptive message here to help people diagnose the problem?

Comment (pervazea) [reply src/network/peloton_server.cpp]: Yes, the message could be more helpful. Let me do that in my next PR... more clean up to do.

**Pull request merged**